### ISSUE TITLE
Plex connector doesn't work with music played from "By Folder" view

### ISSUE THREAD

**Issue by dgw** [david-sabata/web-scrobbler#954]
### tl;dr

Plex connector doesn't work in certain situations; lots of HTML included to help figure out why. HTML grabbed from Chrome Inspector view and formatted as minimally as possible with [this online tool](http://www.freeformatter.com/html-formatter.html) (should be just indentation).

Initially I thought it was broken because Plex completely rewrote large chunks of their Web client earlier this month, but it turns out to be a less obvious issue.
### Full Details

If I play a track using the "normal" browse interface to navigate by album/artist/track name, the HTML looks like this, and scrobbling works:

``` HTML
<div class="player mini-player music">
  <div class="show-volume">
    <div class="mini-controls">
      <div class="seek-bar-container">
        <div class="player-seek-bar player-slider">
          <div class="player-slider-track">
            <div class="player-slider-buffer" style="width: 19.9711%;"></div>
            <div class="player-slider-progress" style="width: 20.1612%;"></div>
          </div>
          <div class="player-slider-thumb" style="left: 20.1612%;"></div>
        </div>
      </div>
      <div class="mini-controls-left">
        <a class="media-poster-container clickable" href="#">
          <div class="media-poster loaded" data-image-title="THE IDOLM@STER ANIM@TION MASTER 01" data-title="READY!! (M@STER VERSION)" data-parent-title="THE IDOLM@STER ANIM@TION MASTER 01" data-grandparent-title="765PRO ALLSTARS" data-image-url="https://10-0-0-90.hash-scrubbed.plex.direct:32400/photo/:/transcode?url=http%3A%2F%2F127.0.0.1%3A32400%2Flibrary%2Fmetadata%2F14182%2Fthumb%2F1461360106%3FX-Plex-Token%3Dtoken-scrubbed&amp;width=80&amp;height=80&amp;minSize=1&amp;X-Plex-Token=token-scrubbed" style="background-image: url(&quot;blob:https%3A//app.plex.tv/3c1600db-851f-4921-803f-d860522016cf&quot;);"></div>
        </a>
      </div>
      <div class="mini-player-scrub-preview persistent-tooltip top fade" style="left: 429px;">
        <div class="tooltip-arrow"></div>
        <div class="tooltip-inner">1:32</div>
      </div>
      <div class="mini-controls-right">
        <div class="mini-controls-below-track-segment media-title">
          <div class="vendor-logo vendor-logo-mixradio"></div>
          <div class="grandparent-title-container"> <button class="grandparent-title btn-link btn-metadata-link clickable">765PRO ALLSTARS</button> </div>
          <div class="title-container"> <button class="item-title btn-link btn-metadata-link clickable">READY!! (M@STER VERSION)</button> </div>
        </div>
        <div class="mini-controls-below-track-segment mini-controls-numeric-progress"> <span class="player-position">0:53</span><span class="separator">/</span><span class="player-duration">4:23</span> </div>
        <div class="mini-controls-below-track-segment">
          <button class="previous-btn btn-link disabled"><i class="glyphicon fast-backward"></i></button> <button class="play-btn btn-link btn-link-lg hidden"><i class="glyphicon play"></i></button> <button class="pause-btn btn-link btn-link-lg"><i class="glyphicon pause"></i></button> <button class="stop-btn btn-link btn-link-lg"><i class="glyphicon stop"></i></button> <button class="next-btn btn-link"><i class="glyphicon fast-forward"></i></button> 
          <span id="player-actions-dropdown" class="dropup">
            <button class="more-btn btn-link dropdown-toggle" href="#player-actions-dropdown" data-toggle="dropdown"><i class="glyphicon more"></i></button> 
            <ul class="dropdown-menu">
              <li><a class="play-mix-btn" href="#" tabindex="-1">Queue Plex Mix</a></li>
              <li><a class="add-to-playlist-btn" href="#" tabindex="-1">Add to Playlist...</a></li>
              <li class="divider"></li>
              <li><a class="sync-btn" href="#" tabindex="-1">Sync...</a></li>
              <li><a class="download-btn" href="https://10-0-0-90.hash-scrubbed.plex.direct:32400/library/parts/17362/file.flac?X-Plex-Token=token-scrubbed" download="01 - READY!! (M@STER VERSION).flac" tabindex="-1">Download</a></li>
              <li><a class="share-btn hidden" href="#" tabindex="-1">Share...</a></li>
              <li><a class="recommend-btn hidden" href="#" tabindex="-1">Recommend...</a></li>
              <li><a class="info-btn" href="#" tabindex="-1">Info</a></li>
            </ul>
          </span>
        </div>
        <div class="mini-controls-below-track-segment player-dropdowns-container rating-container"><span class="rating star-rating unrated"><span class="star"><i class="star-icon glyphicon dislikes"></i><span class="empty-icon"></span></span><span class="star"><i class="star-icon glyphicon dislikes"></i><span class="empty-icon"></span></span><span class="star"><i class="star-icon glyphicon dislikes"></i><span class="empty-icon"></span></span><span class="star"><i class="star-icon glyphicon dislikes"></i><span class="empty-icon"></span></span><span class="star"><i class="star-icon glyphicon dislikes"></i><span class="empty-icon"></span></span> </span></div>
        <div class="mini-controls-below-track-segment ancillary-btns">
          <span class="player-lyrics-container">
            <div id="lyrics-dropdown" class="dropup">
              <button class="lyrics-btn btn-link dropdown-toggle disabled" data-toggle="dropdown" href="#lyrics-dropdown"> <i class="plex-icon-lyrics"></i> </button> 
              <div class="lyrics-dropdown dropdown-menu pull-center"></div>
            </div>
          </span>
          <button class="shuffle-btn btn-link"><i class="plex-icon-osd-shuffle"></i></button> <button class="repeat-btn btn-link"> <i class="repeat-one plex-icon-osd-repeat-one"></i> <i class="repeat plex-icon-osd-repeat"></i> </button> 
        </div>
        <div class="mini-controls-below-track-segment">
          <button class="volume-btn btn-link low"> <i class="glyphicon volume-up"></i> <i class="glyphicon volume-down"></i> <i class="glyphicon mute"></i> </button> 
          <span class="player-volume-slider-container">
            <div class="player-volume-slider player-slider">
              <div class="player-slider-track">
                <div class="player-slider-progress" style="width: 7%;"></div>
              </div>
              <div class="player-slider-thumb" style="left: 7%;"></div>
            </div>
          </span>
          <button class="enter-fullscreen-btn fullscreen-btn btn-link hidden"><i class="glyphicon resize-full"></i></button> <button class="exit-fullscreen-btn fullscreen-btn btn-link hidden"><i class="glyphicon resize-small"></i></button> <button class="maximize-btn btn-link hidden"><i class="glyphicon chevron-up"></i></button> 
          <span class="player-play-queue-container">
            <span id="player-play-queue-dropdown" class="dropup">
              <button class="btn-link dropdown-toggle " href="#play-queue-dropdown" data-toggle="dropdown"> <i class="glyphicon show-thumbnails-with-lines"></i> </button> 
              <ol class="play-queue-list dropdown-menu pull-right ">
                <li class="dropdown-header play-queue-header">  <button class="btn-gray save-btn" href="#"> <i class="plex-icon-add-to-playlist"></i> </button>  <i class="glyphicon show-thumbnails-with-lines"></i> Up Next </li>
                <li data-play-queue-id="1952" class="play-queue-list-item">
                  <a class="play-queue-metadata-region" href="#">
                    <div>
                      <div class="media-poster track loaded" style="background-image: url(&quot;blob:https%3A//app.plex.tv/0c20e645-ef60-4e93-9b48-6389e6f38685&quot;);"></div>
                      <div class="media-item">
                        <div class="media-title">おとなのはじまり</div>
                        <div class="media-subtitle-1"><span class="media-primary-subtitle">765PRO ALLSTARS</span><span class="media-secondary-subtitle"> - THE IDOLM@STER ANIM@TION MASTER 01</span></div>
                      </div>
                      <button class="delete-btn btn-link btn-danger" tabindex="-1"><i class="glyphicon remove-2"></i></button>  
                    </div>
                  </a>
                </li>
                <li data-play-queue-id="1953" class="play-queue-list-item">
                  <a class="play-queue-metadata-region" href="#">
                    <div>
                      <div class="media-poster track loaded" style="background-image: url(&quot;blob:https%3A//app.plex.tv/0c20e645-ef60-4e93-9b48-6389e6f38685&quot;);"></div>
                      <div class="media-item">
                        <div class="media-title">おもいでのはじまり</div>
                        <div class="media-subtitle-1"><span class="media-primary-subtitle">765PRO ALLSTARS</span><span class="media-secondary-subtitle"> - THE IDOLM@STER ANIM@TION MASTER 01</span></div>
                      </div>
                      <button class="delete-btn btn-link btn-danger" tabindex="-1"><i class="glyphicon remove-2"></i></button>  
                    </div>
                  </a>
                </li>
                <li data-play-queue-id="1954" class="play-queue-list-item">
                  <a class="play-queue-metadata-region" href="#">
                    <div>
                      <div class="media-poster track loaded" style="background-image: url(&quot;blob:https%3A//app.plex.tv/0c20e645-ef60-4e93-9b48-6389e6f38685&quot;);"></div>
                      <div class="media-item">
                        <div class="media-title">READY!! (M@STER VERSION) (Instrumental)</div>
                        <div class="media-subtitle-1"><span class="media-primary-subtitle">765PRO ALLSTARS</span><span class="media-secondary-subtitle"> - THE IDOLM@STER ANIM@TION MASTER 01</span></div>
                      </div>
                      <button class="delete-btn btn-link btn-danger" tabindex="-1"><i class="glyphicon remove-2"></i></button>  
                    </div>
                  </a>
                </li>
              </ol>
            </span>
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
```

If I navigate to the same file using the "By Folder" view, the HTML looks like this, and scrobbling does **not** work:

``` HTML
<div class="player mini-player music">
  <div class="show-volume">
    <div class="mini-controls">
      <div class="seek-bar-container">
        <div class="player-seek-bar player-slider">
          <div class="player-slider-track">
            <div class="player-slider-buffer" style="width: 5.55125%;"></div>
            <div class="player-slider-progress" style="width: 5.4984%;"></div>
          </div>
          <div class="player-slider-thumb" style="left: 5.4984%;"></div>
        </div>
      </div>
      <div class="mini-controls-left">
        <a class="media-poster-container clickable" href="#">
          <div class="media-poster placeholder-square" data-image-title="READY!! (M@STER VERSION)" data-title="READY!! (M@STER VERSION)"></div>
        </a>
      </div>
      <div class="mini-player-scrub-preview persistent-tooltip top fade" style="left: 361px;">
        <div class="tooltip-arrow"></div>
        <div class="tooltip-inner">1:18</div>
      </div>
      <div class="mini-controls-right">
        <div class="mini-controls-below-track-segment media-title">
          <div class="vendor-logo vendor-logo-mixradio"></div>
          <div class="grandparent-title-container"> <button class="grandparent-title btn-link btn-metadata-link clickable hidden"></button> </div>
          <div class="title-container"> <button class="item-title btn-link btn-metadata-link clickable">READY!! (M@STER VERSION)</button> </div>
        </div>
        <div class="mini-controls-below-track-segment mini-controls-numeric-progress"> <span class="player-position">0:14</span><span class="separator">/</span><span class="player-duration">4:23</span> </div>
        <div class="mini-controls-below-track-segment">
          <button class="previous-btn btn-link disabled"><i class="glyphicon fast-backward"></i></button> <button class="play-btn btn-link btn-link-lg hidden"><i class="glyphicon play"></i></button> <button class="pause-btn btn-link btn-link-lg"><i class="glyphicon pause"></i></button> <button class="stop-btn btn-link btn-link-lg"><i class="glyphicon stop"></i></button> <button class="next-btn btn-link"><i class="glyphicon fast-forward"></i></button> 
          <span id="player-actions-dropdown" class="dropup">
            <button class="more-btn btn-link dropdown-toggle" href="#player-actions-dropdown" data-toggle="dropdown"><i class="glyphicon more"></i></button> 
            <ul class="dropdown-menu">
              <li><a class="play-mix-btn hidden" href="#" tabindex="-1">Play Plex Mix</a></li>
              <li><a class="add-to-playlist-btn" href="#" tabindex="-1">Add to Playlist...</a></li>
              <li class="divider"></li>
              <li><a class="sync-btn" href="#" tabindex="-1">Sync...</a></li>
              <li><a class="download-btn" href="https://10-0-0-90.hash-scrubbed.plex.direct:32400/library/parts/17362/file.flac?X-Plex-Token=token-scrubbed" download="01 - READY!! (M@STER VERSION).flac" tabindex="-1">Download</a></li>
              <li><a class="share-btn hidden" href="#" tabindex="-1">Share...</a></li>
              <li><a class="recommend-btn hidden" href="#" tabindex="-1">Recommend...</a></li>
              <li><a class="info-btn" href="#" tabindex="-1">Info</a></li>
            </ul>
          </span>
        </div>
        <div class="mini-controls-below-track-segment player-dropdowns-container rating-container"><span class="rating star-rating unrated"><span class="star"><i class="star-icon glyphicon dislikes"></i><span class="empty-icon"></span></span><span class="star"><i class="star-icon glyphicon dislikes"></i><span class="empty-icon"></span></span><span class="star"><i class="star-icon glyphicon dislikes"></i><span class="empty-icon"></span></span><span class="star"><i class="star-icon glyphicon dislikes"></i><span class="empty-icon"></span></span><span class="star"><i class="star-icon glyphicon dislikes"></i><span class="empty-icon"></span></span> </span></div>
        <div class="mini-controls-below-track-segment ancillary-btns">
          <span class="player-lyrics-container">
            <div id="lyrics-dropdown" class="dropup">
              <button class="lyrics-btn btn-link dropdown-toggle disabled" data-toggle="dropdown" href="#lyrics-dropdown"> <i class="plex-icon-lyrics"></i> </button> 
              <div class="lyrics-dropdown dropdown-menu pull-center"></div>
            </div>
          </span>
          <button class="shuffle-btn btn-link disabled"><i class="plex-icon-osd-shuffle"></i></button> <button class="repeat-btn btn-link"> <i class="repeat-one plex-icon-osd-repeat-one"></i> <i class="repeat plex-icon-osd-repeat"></i> </button> 
        </div>
        <div class="mini-controls-below-track-segment">
          <button class="volume-btn btn-link low"> <i class="glyphicon volume-up"></i> <i class="glyphicon volume-down"></i> <i class="glyphicon mute"></i> </button> 
          <span class="player-volume-slider-container">
            <div class="player-volume-slider player-slider">
              <div class="player-slider-track">
                <div class="player-slider-progress" style="width: 7%;"></div>
              </div>
              <div class="player-slider-thumb" style="left: 7%;"></div>
            </div>
          </span>
          <button class="enter-fullscreen-btn fullscreen-btn btn-link hidden"><i class="glyphicon resize-full"></i></button> <button class="exit-fullscreen-btn fullscreen-btn btn-link hidden"><i class="glyphicon resize-small"></i></button> <button class="maximize-btn btn-link hidden"><i class="glyphicon chevron-up"></i></button> 
          <span class="player-play-queue-container">
            <span id="player-play-queue-dropdown" class="dropup">
              <button class="btn-link dropdown-toggle " href="#play-queue-dropdown" data-toggle="dropdown"> <i class="glyphicon show-thumbnails-with-lines"></i> </button> 
              <ol class="play-queue-list dropdown-menu pull-right ">
                <li class="dropdown-header play-queue-header">  <button class="btn-gray save-btn" href="#"> <i class="plex-icon-add-to-playlist"></i> </button>  <i class="glyphicon show-thumbnails-with-lines"></i> Up Next </li>
                <li data-play-queue-id="6" class="play-queue-list-item">
                  <a class="play-queue-metadata-region" href="#">
                    <div>
                      <div class="media-poster track placeholder-square"></div>
                      <div class="media-item">
                        <div class="media-title">おとなのはじまり</div>
                        <div class="media-subtitle-1"><span class="media-primary-subtitle"></span><span class="media-secondary-subtitle"></span></div>
                      </div>
                      <button class="delete-btn btn-link btn-danger" tabindex="-1"><i class="glyphicon remove-2"></i></button>  
                    </div>
                  </a>
                </li>
                <li data-play-queue-id="7" class="play-queue-list-item">
                  <a class="play-queue-metadata-region" href="#">
                    <div>
                      <div class="media-poster track placeholder-square"></div>
                      <div class="media-item">
                        <div class="media-title">おもいでのはじまり</div>
                        <div class="media-subtitle-1"><span class="media-primary-subtitle"></span><span class="media-secondary-subtitle"></span></div>
                      </div>
                      <button class="delete-btn btn-link btn-danger" tabindex="-1"><i class="glyphicon remove-2"></i></button>  
                    </div>
                  </a>
                </li>
                <li data-play-queue-id="8" class="play-queue-list-item">
                  <a class="play-queue-metadata-region" href="#">
                    <div>
                      <div class="media-poster track placeholder-square"></div>
                      <div class="media-item">
                        <div class="media-title">READY!! (M@STER VERSION) (Instrumental)</div>
                        <div class="media-subtitle-1"><span class="media-primary-subtitle"></span><span class="media-secondary-subtitle"></span></div>
                      </div>
                      <button class="delete-btn btn-link btn-danger" tabindex="-1"><i class="glyphicon remove-2"></i></button>  
                    </div>
                  </a>
                </li>
              </ol>
            </span>
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
```

**Comment by alexesprit** [comment]
It' not possible to fix due to Plex UI limitations. The solution is to listen to music via https://app.plex.tv/web/.
